woocommerce/woocommerce
# dev: added xstate inspector toggling

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds a toggle for XState visual inspector.

To enable the XState inspector, open the browser console and type:
 
`localStorage.setItem("xstate_inspect", "true")`

Also added eslint-plugin-xstate for xstate lint rules.



### How to test the changes in this Pull Request:

Dev tooling only, no testing needed


1. Run a local build, enter this in your browser console: `localStorage.setItem("xstate_inspect", "true")`
2. Visit a page that has a XState machine on it, such as http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
3. Should see the inspector show up in a new tab
4. Should also see lint violation warnings in  `./client/core-profiler/index.tsx` , vscode eslint extension recommended as there are maaaaaany warnings in the whole codebase and it is hard to notice any lint violations.
